Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +27,123 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        f = formula do
+          url "foo-1.0"
+          def plist
+            "plist_test.plist"
+          end
+          plist_options startup: true
+        end
+        expect(described_class.new(f).caveats).to include("startup:\n  sudo brew")
+      end
+
+      it "prints plist login information when f.plist_startup is nil" do
+        f = formula do
+          url "foo-1.0"
+          def plist
+            "plist_test.plist"
+          end
+        end
+        expect(described_class.new(f).caveats).to include("login:\n  brew")
+      end
+    end
+
+    context "when f.keg_only is not nil" do
+      Path = Pathname.new("path")
+
+      ## Refactoring needed:
```

Comment:
Use `let(:f) { formula { ... } }` in each of the `context`s to set `f` for all the values in the context.